compose/cli/main.py

```diff
@@ -655,20 +655,19 @@ def remove_container(force=False):
         if options['--rm']:
             project.client.remove_container(container.id, force=True)
 
-    def force_shutdown(signal, frame):
-        project.client.kill(container.id)
-        remove_container(force=True)
-        sys.exit(2)
-
-    def shutdown(signal, frame):
-        set_signal_handler(force_shutdown)
-        project.client.stop(container.id)
-        remove_container()
-        sys.exit(1)
+    signals.set_signal_handler_to_shutdown()
+    try:
+        dockerpty.start(project.client, container.id, interactive=not options['-T'])
+        exit_code = container.wait()
```

Comment:
The logic is not repeated. `up` and `run` do **different** things. The shutdown operation is different.  

We gain nothing by moving them together, it would actually make the code harder to follow because we'd have unrelated shutdown happening in the same function.